src/Disks/ObjectStorages/DiskObjectStorageMetadata.cpp

```diff
@@ -131,9 +131,6 @@ DiskObjectStorageMetadata::DiskObjectStorageMetadata(
 
 void DiskObjectStorageMetadata::addObject(const String & path, size_t size)
 {
-    if (!object_storage_root_path.empty() && path.starts_with(object_storage_root_path))
```

Comment:
blobs paths are generated random strings, so if user uses root path for all objects as a small simple string, it is possible that object path prefix is the same as root_path, as a result a logical error will be thrown even though everything is correct.